src/macaron/slsa_analyzer/package_registry/deps_dev.py

```diff
@@ -20,28 +21,24 @@ class DepsDevService:
     """The deps.dev service class."""
 
     @staticmethod
-    def get_package_info(purl: str) -> dict | None:
-        """Check if the package identified by the PackageURL (PURL) exists and return its information.
+    def get_endpoint(purl: bool = True, path: str | None = None) -> urllib.parse.SplitResult:
```

Comment:
I have a couple of comments about this function
- Because we have two types of end point for different purposes, the PURL endpoint `https://api.deps.dev/v3alpha/purl/` and the BASE endpoint `https://api.deps.dev/v3alpha/`. I wonder if we could separate them into 2 functions?
```
def get_base_endpoint(...)
def get_purl_endpoint(...)
```
- Could we add a small comment (may be as a doc string) to explicitly say that `path` is added directly to the end of the URL endpoint without any modification? - This is just to prevent using this method without percent encoding `path` before hand.